### PR TITLE
Fixes #106: Sets line-height to 0 (except in label) and --paper-checkbox-label-spacing to 0 in tests.

### LINKED ISSUE #106: Tests are failing
#94 fixed misused comparison operators in a few assertions, revealing that the tests they were part of actually failed.

### PR DESCRIPTION
**Pull request by bicknellr**
Fixes #106

### CODE REVIEW

File: paper-checkbox.html [hunk 1]
```diff
@@ -171,6 +172,7 @@
         vertical-align: middle;
         padding-left: var(--paper-checkbox-label-spacing, 8px);
         white-space: normal;
+        line-height: normal;
```
Comment (notwaldorf) [inline paper-checkbox.html]: Did you test that if the user sets a different line height on the host (say `line-height: 2`), this doesn't break it? I.e. Something like this: http://jsbin.com/vuyosa/edit?html,output

Comment (bicknellr) [inline paper-checkbox.html]: http://jsbin.com/ruveye/edit?html,output Seems ok; line-height is relative to font-size and [the spec](https://www.w3.org/TR/CSS2/visudet.html#propdef-line-height) kind of implies that `normal` is a constant. Also, line-height is apparently a minimum and contents of the line can expand this, so `normal` should cause the checkbox to expand to show the text regardless of what line-height the user sets on the host. Do you think this should have an actual test?

Comment (notwaldorf) [inline paper-checkbox.html]: You're setting this on the checkboxLabel though, not the checkbox itself, but it looks right!

**Comment by notwaldorf** [comment]
LGTM

**Comment by bicknellr** [comment]
@notwaldorf I added a couple tests.

**Comment by notwaldorf** [comment]
Thanks for the extra tests! Still LGTM

**Pull request merged**